tests/lambda_tests/test_decode.py

```diff
@@ -0,0 +1,45 @@
+import unittest
+from src.decode_function import BucketReader, BucketReaderError
+from unittest.mock import Mock
+
+
+class TestBucketReader(unittest.TestCase):
+    def setUp(self):
+        self.mock_client = Mock()
+        self.bucket_reader = BucketReader(self.mock_client)
+        self.object_content = Mock()
+        self.object_data = Mock()
+        self.object_content.read.return_value = bytes('content', 'utf-8')
+        self.dictionary = {'Body': self.object_content}
+
+    def test_reader(self):
+        self.bucket_reader.s3.get_object.return_value = self.dictionary
+        self.bucket_reader.read_object_content(bucket="bucket", object_key="key")
+        self.bucket_reader.s3.get_object.assert_called_once_with(Bucket="bucket", Key="key")
+
+    def test_get_object_side_effect(self):

```

Comment:
side effect is too broad. A better name would be `test_get_object_exception_handling`